react/features/base/participants/functions.js

```diff
@@ -484,11 +484,19 @@ export function shouldRenderParticipantVideo(stateful: Object | Function, id: st
         return false;
     }
 
-    /* Then check if the participant connection is active. */
-    const connectionStatus = participant.connectionStatus || JitsiParticipantConnectionStatus.ACTIVE;
+    /* Then check if the participant connection or track streaming status is active. */
+    if (getSourceNameSignalingFeatureFlag(state)) {
```

Comment:
It's worth leaving a TODO/comment that this code will only work if there's something else registered as a listener of this specific `jitsiTrack`s streaming status (it's disabled when there's 0 listeners).